rotkehlchen/exchanges/okx.py

```diff
@@ -0,0 +1,498 @@
+            amount = deserialize_asset_amount(raw_trade['accFillSz'])
+            rate = deserialize_price(raw_trade['avgPx'])
+            fee_amount = deserialize_fee(raw_trade['fee'])
+            # fee charged by the platform is represented by a negative number
+            if fee_amount < 0:
+                fee_amount = Fee(-1 * fee_amount)
+            fee_asset = asset_from_okx(raw_trade['feeCcy'])
+
+            return Trade(
+                timestamp=timestamp,
+                location=Location.OKX,
+                base_asset=base_asset,
+                quote_asset=quote_asset,
+                trade_type=trade_type,
+                amount=amount,
+                rate=rate,
+                fee=fee_amount,
+                fee_currency=fee_asset,
+                link=raw_trade['ordId'],
+            )
+        except UnknownAsset as e:
+            self.msg_aggregator.add_warning(
+                f'Found {self.name} trade with unknown asset '
+                f'{e.identifier}. Ignoring it.',
+            )
+        except UnsupportedAsset as e:
+            self.msg_aggregator.add_warning(
+                f'Found {self.name} trade with unsupported asset '
+                f'{e.identifier}. Ignoring it.',
+            )
+        except (DeserializationError, KeyError) as e:
+            self.msg_aggregator.add_error(
+                f'Unexpected data encountered during deserialization of {self.name}'
+                'trade. Check logs for details and open a bug report.',
+            )
+            msg = str(e)
+            if isinstance(e, KeyError):
+                msg = f'Missing key entry for {msg}.'
+            log.error(
+                f'Unexpected data encountered during deserialization of {self.name} '
+                f'trade {raw_trade}. Error was: {msg}',
+            )
+
+        return None
+
+    def asset_movement_from_okx(
+            self,
+            raw_movement: dict[str, Any],
+            category: AssetMovementCategory,
+    ) -> Optional[AssetMovement]:
+        """
+        Converts a raw asset movement from OKX into an AssetMovement object.
+        If there is an error `None` is returned and error is logged.
+        """
+        try:
+            tx_hash = raw_movement['txId']
+            timestamp = ts_ms_to_sec(TimestampMS(int(raw_movement['ts'])))
+            asset = asset_from_okx(raw_movement['ccy'])
+            amount = deserialize_asset_amount(raw_movement['amt'])
+            address = deserialize_asset_movement_address(raw_movement, 'to', asset)
+            fee = Fee(ZERO)
+            if category is AssetMovementCategory.WITHDRAWAL:
+                fee = deserialize_fee(raw_movement['fee'])
+        except UnknownAsset as e:
+            self.msg_aggregator.add_warning(
+                f'Found {self.name} deposit/withdrawal with unknown asset '
+                f'{e.identifier}. Ignoring it.',
+            )
+            return None
```

Comment:
ah I see that you need to repeat the `None` this way. You can fix this by doing something like `asset_movement = None` at the start of the function, at success assign it, and at the very end return it.

So you don't need to repeat `return None` in each except.